arraycontext/container/arithmetic.py

```diff
@@ -343,6 +343,17 @@ def {fname}(arg1):
                                 else:
                                     raise ValueError(msg)""")
                     gen(f"return cls({zip_init_args})")
+                if _cls_has_array_context_attr:
+                    gen("if isinstance(arg2,"
+                        "              arg1.array_context.get_array_types()):")
```

Comment:
(I can't seem to comment on the same line multiple times, hence more comments on subsequent lines.) `isinstance` is rather expensive. Can we get by with `arg2.__class__ is ...`?

Reply:
That would be pretty problematic as far as pytato is concerned. There isn't one single array class.

Reply:
Good point.